### DISCUSSION TITLE
Problem: longer conversations, slower generations.

### DISCUSSION THREAD

**Discussion by EugeoSynthesisThirtyTwo** [oobabooga/text-generation-webui#350] [Ideas]
Sorry if my english contains mistakes, I am french.

I am using the webui in --cai-chat mode. I noticed that at the beginning of the conversation the generation is fast (~6 tokens/s), and after 30 messages (including my owns) the generation is super slow (0.3 tokens/s).

Do you know where it could come from? Is it because the model has to be feed with the entire conversation on each generation? If that is the case then I would appreciate a way to run the chat mode without the possibility of editing/removing last messages but instead only the last message would be feed into the model on each generation.

I have windows 10, I have an RTX 3080 TI laptop GPU (16 GB VRAM), and a CPU with 64 GB RAM, and my command line is the following:
`python server.py --auto-devices --cai-chat --model llama-13b --gptq-bits 4 --gpu-memory 15 --listen`

Please let me know if I should move this thread into issues.

**Comment by patrickmros** [comment]
> I would appreciate a way to run the chat mode without the possibility of editing/removing last messages but instead only the last message would be feed into the model on each generation

Wouldn't that be a chat bot with no short term memory? If only the last message is fed back, the AI would have almost no context.

**Reply by EugeoSynthesisThirtyTwo** [reply]
I can't answer that question sorry, I don't really know how these models work. I was just asking if there was a way to fix the generation slowing down

**Comment by keldenl** [comment]
I was just thinking about this -- what about if instead every 5 messages or so an additional request is made to summarize the previous 5 messages?

that way, long messages back and forths wont be as bad at a cost of extra processing every 5 messages for example

**Comment by jepjoo** [comment]
Am I tripping or is this effect much more pronounced with the 4-bit versions?

I'd been using the LLaMa-13B in 8-bit for several days and didn't notice a slowdown at any point (maybe subtle enough?). First couple of tries with 13B in 4-bit and the slowdown feels quite dramatic after like 10-20 chat messages.

**Reply by EugeoSynthesisThirtyTwo** [reply]
I have this problem with the 4 bits version too but llama-7b 8-bit works fine

**Comment by nullgate** [comment]
Check your config.json in the model directory.
Make sure that there is this line

`    "use_cache": true,`

It's very slow if use_cache is not enabled. It's also disabled with --no-cache parameter.

Without cache, the model must recompute everything from input_ids, which is prohibitively slow for longer context/output.
The best way to understand this is to read the code.

**Reply by oobabooga** [reply]
Based source code reader.

**Comment by oobabooga** [comment]
I think that's a limitation of the transformers architecture. The generation does become slower as the context size increases.
